merge changes from upstream

sdtblck opened the pull request.

tjruwase committed:
Refactor moe/non-moe gradient reduction (#1811)

andriyor committed:
add GitHub URL for PyPi (#1812)

* add GitHub URL for PyPi

* add GitHub URL for PyPi fix formatting

jithunnair-amd committed:
Invoke hipify from op builder for JIT extension builds (#1807)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
force amd install via sudo (#1815)

jeffra committed:
bump to 0.6.1

jeffra committed:
update gemfile (#1817)

jeffra committed:
[docker] simplify and update rocm dockerfile (#1819)

hyunwoongko commented:
@sdtblck Can you list the features of DeeperSpeed that you are currently using on NeoX? We will implement it sequentially in OSLO. Perhaps the most important feature is ZeRO optimization, but if you need additional features, please let us know.

EugeneLYC committed:
01 adam optimizer (#1790)

Co-authored-by: Conglong Li <conglong.li@gmail.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

rraminen committed:
Enable CG headers on ROCm (#1821)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

cli99 committed:
Website posts and tutorial improvements (#1799)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
[website] remove extra news header

cli99 committed:
Tune webpage width (#1829)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

tjruwase committed:
Reduce specified gradients (#1818)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
[launcher] validate passwordless-ssh works when using hostfile launching (#1832)

mrwyattii committed:
Split github action unit tests by platform/GPU (#1828)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

stas00 committed:
[build] support cuda-11.6 (#1836)

to support the new cuda

jeffra committed:
[ZeRO-1] fix bug w. cpu-offload + > 1 GPU (#1841)

rraminen committed:
THCGeneral.h header file is deprecated (#1842)

Co-authored-by: hubertlu-tw <hubertlu@amd.com>

shjwudp committed:
fix: fix undefined variable in MoE top2gating (#1827)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

shjwudp committed:
fix: Fix undefined variable in _create_expert_data_and_model_parallel and make it easier to understand (#1826)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

tjruwase committed:
Track only trainable parameters (#1780)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

mrwyattii committed:
Add concurrency policy to CI workflow (#1844)

* added concurrency to github actions

* fixed problem where one workflow can cancel another workflow

* added tmp file to help test concurrency policy on CI

* removed tmp file to finish testing concurrency policy

jeffra committed:
add amd release blog (#1848)

awan-10 committed:
deepscale --> deepspeed in prints. (#1854)

xldrx committed:
Update config-json.md (#1853)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
fix dead MoQ link (#1855)

tirkarthi committed:
Import ABC from collections.abc for Python 3.10 compatibility. (#1851)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

samyam committed:
qkv_out can be a single tensor or a list. Handling these cases separetely. (#1850)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
bump to 0.6.2

PaperclipBadger committed:
Fix setup.py crash when torch is not installed. (#1866)

Co-authored-by: Blaine Rogers <blaine.rogers@five.ai>

matherit committed:
Add support for AWS SageMaker. (#1868)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

tjruwase committed:
Fix broken links (#1873)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
[docs] add amd blog to website (#1874)

jeffra committed:
[docs] add moe paper (#1875)

samyam committed:
Supporting multiple modules injection with a single policy when they have identical architectures (#1869)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
[docs] fix dead links (#1877)

stas00 committed:
add now required `-lcurand` to solve `undefined symbol: curandCreateGenerator` (#1879)

VisionTheta committed:
remove the print line in _einsum_flops_compute function (#1885)

dependabot[bot] committed:
Bump nokogiri from 1.13.3 to 1.13.4 in /docs (#1889)

Bumps [nokogiri](https://github.com/sparklemotion/nokogiri) from 1.13.3 to 1.13.4.
- [Release notes](https://github.com/sparklemotion/nokogiri/releases)
- [Changelog](https://github.com/sparklemotion/nokogiri/blob/v1.13.4/CHANGELOG.md)
- [Commits](https://github.com/sparklemotion/nokogiri/compare/v1.13.3...v1.13.4)

---
updated-dependencies:
- dependency-name: nokogiri
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jeffra committed:
[docs] fix commonmarker security issue (#1892)

tjruwase committed:
bf16+pipeline parallelism (#1801)

* bf16 updates

* Got bf16 working

* fp32 reduction; flattened tensors

* bf16+zero_stage_1 first cut

* finish zero_stage 1 sharding

* Matching fp16 with debugging codes

* Matching loss with fp16

* Fix gradient clipping

* bf16 gradient clipping fix
bf16 checkpoint save/load

* Unscale grad norm

* Fix grad norm scaling

* Enable loading fp16_zero_1 into bf16_zero_1 engine and vice versa

* Fix clip_grad key error

* Reduce tied weight gradients

* Fix grad norm for moe

* Reduce specified gradients

* Use O(n) instead of O(n^2)

* Remove optimizer restriction for bf16

* Link bf16 & fp32 params

* Clip gradients of last stage tied weights

* Simplify tied weights reduction logic

* Also clip all tp rank parameters

* lp to hp mapping

* Link lp/hp/optim state; Refresh links after checkpoint load

* Remove debug print

* Remove debug print

* Simplify zero_grad logic

* fp32 accessors

* Fix update bug

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

szhengac committed:
fix file ordering (#1822)

Co-authored-by: Shuai Zheng <shzheng@amazon.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

manuelciosici committed:
Use f-strings where possible (#1900)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

stas00 committed:
[partition_parameters.py] better diagnostics (#1887)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

conglongli committed:
cast bool when not supported by torch2cupy (#1894)

tjruwase committed:
Use cuda events to improve timing for multi-stream execution (#1881)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

tjruwase committed:
Fix multiple zero 3 tracing errors (#1901)

* Fix zero3 tracing issues

* Remove debug prints

* Code clarity

mrwyattii committed:
Improve ds_report output for HIP/ROCm (#1906)

szhengac committed:
fix launcher for reading env vars (#1907)

Co-authored-by: Shuai Zheng <shzheng@amazon.com>
Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

tjruwase committed:
Fix OOM and type mismatch (#1884)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
Inference PP changes for neox (#1899)

Co-authored-by: Reza Yazdani <reyazda@microsoft.com>
Co-authored-by: Reza Yazdani <44502768+RezaYazdaniAminabadi@users.noreply.github.com>

jeffra committed:
[zero-3] add bwd support for list/dict types returned in fwd (#1857)

mrwyattii committed:
bumped to v0.6.3

jeffra committed:
bump to 0.6.4

mrwyattii committed:
lazy import fcntl (#1921)

jeffra committed:
explictly add op_builder to manifest (#1920)

rraminen committed:
Enable DeepSpeed inference on ROCm  (#1922)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

tjruwase committed:
bf16 inference (#1917)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

kisseternity committed:
spell err (#1929)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
[ZeRO-3] Rename confusing log message (#1932)

itszyin committed:
Fix time error (#1934)

jeffra committed:
bump to 0.6.5

tjruwase committed:
Improve z3 trace management (#1916)

* Fix OOM and type mismatch

* Toggle prefetching

* Disable z3 prefetching for inference (temp workaround)

* Fix zero3 tracing issues

* Remove debug prints

* Enable prefetch for inference

* Code clarity

* Invalidate trace cache

* Trace cache invalidation when needed
Separate nvme prefetch from all-gather prefetch

* Track last used step id

* Use debug name in error message

* Construct param trace from module trace

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

stas00 committed:
GatheredParameters - accept a tuple of params (#1941)

* GatheredParameters - accept any iterable

* torch tensor is an iterable, so can't use collections.abc.Iterable

* fix

manuelciosici committed:
Update partition_parameters.py (#1943)

szhengac committed:
fix step in adam (#1823)

* fix step in adam

* fix backward compatibility and add unittest

* add unittest

* fix unbounded error when there are more than 1 param groups

* fix typo

* remove trailing whitespace

* fix end of file

Co-authored-by: Shuai Zheng <shzheng@amazon.com>
Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

stas00 committed:
[pipe] prevent deadlock with multiple evals sequence (#1944)

* [pipe] prevent deadlock with multiple evals sequence

* style

* style

* style

* align DSE commit w. latest master

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>
Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

jeffra committed:
Fairseq support (#1915)

tjruwase committed:
DeepSpeed needs to start cleaning up (#1947)

* DeepSpeed needs to start cleaning up

* Remove debug prints

kisseternity committed:
trivial fix (#1954)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

RezaYazdaniAminabadi committed:
Enabling CUDA-graph for the bert-type models (#1952)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

Quentin-Anthony committed:
Add loss scale guard to avoid inf loop (#1958)

liamcli committed:
[launcher] add option to bypass ssh check (#1957)

dependabot[bot] committed:
Bump nokogiri from 1.13.4 to 1.13.6 in /docs (#1965)

Bumps [nokogiri](https://github.com/sparklemotion/nokogiri) from 1.13.4 to 1.13.6.
- [Release notes](https://github.com/sparklemotion/nokogiri/releases)
- [Changelog](https://github.com/sparklemotion/nokogiri/blob/main/CHANGELOG.md)
- [Commits](https://github.com/sparklemotion/nokogiri/compare/v1.13.4...v1.13.6)

---
updated-dependencies:
- dependency-name: nokogiri
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Quentin-Anthony committed:
Fix timer typo (#1964)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
Update Gemfile.lock (#1966)

jeffra committed:
Update cpu_adam.py (#1968)

Quentin-Anthony committed:
Add Unidirectional Sparse Attention Type to BigBird and BSLongformer (#1959)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

Dipet committed:
Fix: Sparse tensors not updating (#1914)

* Fix do not updated sparse grads

* Remove call .data for sparse grads

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

RezaYazdaniAminabadi committed:
Fixing several bugs in the inference-api and the kernels (#1951)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
bump to 0.6.6

jeffra committed:
add 530b paper (#1979)

RezaYazdaniAminabadi committed:
small fix for the HF Bert models (#1984)

mrwyattii committed:
Add unit test for various model families and inference tasks (#1981)

* added unit test for various HF model families and tasks

* formatting

* added missing import

* fixed broken pytest global vars

* modified test to conform to other test structure

* removed gpt-j. it cannot run on V100s (OOM)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

mrwyattii committed:
added temp fix for our CI jobs (#1988)

cli99 committed:
fix typo when getting kernel dim in conv calculation (#1989)

mrwyattii committed:
Add torch-latest and torch-nightly CI workflows (#1990)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

mrwyattii committed:
[bug] Add user-defined launcher args for MPI launcher (#1933)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jerrymannil committed:
Propagate max errorcode to deepspeed when using PDSH launcher (#1994)

Add '-S' argument to pdsh command to return the largest error code from the ssh sessions

jeffra committed:
[docs] add new build badges to landing page (#1998)

awan-10 committed:
DeepSpeed comm backend v1 (#1985)

Co-authored-by: Quentin Anthony <qganthony@yahoo.com>
Co-authored-by: Ammar Ahmad Awan <ammar.awan@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

Quentin-Anthony committed:
Relax assertion to allow Megatron-DeepSpeed MoE to use ZeRO-1 (#2007)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

conglongli committed:
update CODEOWNERS (#2017)

jeffra committed:
[CI] force upgrade HF dependencies & output py env (#2015)

jeffra committed:
[inference] test suite for ds-kernels (bert, roberta, gpt2, gpt-neo, gpt-j) (#1992)

Co-authored-by: Reza Yazdani <reyazda@microsoft.com>
Co-authored-by: Michael Wyatt <michaelwyatt@microsoft.com>
Co-authored-by: Reza Yazdani <44502768+RezaYazdaniAminabadi@users.noreply.github.com>

jeffra committed:
DeepSpeed examples refresh (#2021)

RezaYazdaniAminabadi committed:
Fix transformer API for training-evaluation pipeline (#2018)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

Sanger2000 committed:
DataLoader Length Fix (#1718)

Quentin-Anthony committed:
DeepSpeed Monitor Module (Master) (#2013)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

tjruwase committed:
Use partition size (#2011)

KMFODA committed:
fix import errors (#2026)

Co-authored-by: Michael Wyatt <michaelwyatt@microsoft.com>

mrwyattii committed:
Fix inference unit test import error catching (#2024)

* fix to catch assert error for inference test imports

* fix wrong syntax

* changed to sequential inf tests

* fix for lm_eval import

* added environment check fixture

* added expected torch and cuda version

* check various version depth for cuda/torch

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

tjruwase committed:
Retain available params until last use (#2016)

* Retain prefetched params until last use

* Unit tests fixes

tjruwase committed:
Split parameter offload from z3 (#2009)

* Split parameter offload from z3

* Format fixes

* Bug fixes

* Cleanup

* Remove dead code

mrwyattii committed:
fixed print statement (#2038)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

conglongli committed:
Add compression papers (#2042)

RezaYazdaniAminabadi committed:
Fix the half-precision version of CPU-Adam (#2032)

* Fix the half-precision version of CPU-Adam

* remove unexpected return

* fix the increase width (fp32/fp16)

* support fp16 tests for cpu-adam

* fix the fp16 data-loading

* change unit-test for fp16 check & slight change to parameter size

* fix for numpy error

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>
Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Michael Wyatt <michaelwyatt@microsoft.com>

mrwyattii committed:
Fix for AMD unit tests (#2047)

* assert no FP16 with AMD CPUs

* add unit test for AMD assert error

* missing import

* downgrade assert to warning

siddharth9820 committed:
correct partition_id in fp32 param -> fp16 param for MoE+z2 (#2058)

aphedges committed:
Fix missing import in replace_module.py (#2050)

* Fix missing import in replace_module.py

* Change import from torch.distributed to deepspeed.comm

Quentin-Anthony committed:
Comms Benchmarks (#2040)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
add ds inference paper (#2072)

kisseternity committed:
Comments for better understanding of zero stage1_2  (#2027)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
[docs] fix broken read-the-docs build (#2075)

aphedges committed:
Check for bf16 support only if CUDA is available (#2049)

Co-authored-by: Michael Wyatt <michaelwyatt@microsoft.com>

siddharth9820 committed:
Fix partition id in the fp32->fp16 param copying step for z2+cpu-offload (#2059)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

samadejacobs committed:
Codeowner addendum and fix to small model debugging script (#2076)

* Add github username to CODEOWNERS

* add import torch.distributed to small model debugging test script

* Replace torch.dist with deepspeed.com

cli99 committed:
remove require grad in params count (#2065)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

manuelciosici committed:
Add missing newline for ZeroOneAdam parameter table (#2088)

Co-authored-by: Conglong Li <conglong.li@gmail.com>

xiazeyu committed:
fixed "None type has no len()" (#2091)

Co-authored-by: Cheng Li <pistasable@gmail.com>

siddharth9820 committed:
Improving memory utilization of Z2+MoE (#2079)

* Shards expert parameter groups
* Do upscaling, optimizer and deletion of fp32 grads one-by-one on each parameter group in zero-2
Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

jeffra committed:
bump to 0.6.7

RezaYazdaniAminabadi committed:
Add Inference support for running the BigScience-BLOOM Architecture (#2083)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>
Co-authored-by: Michael Wyatt <michaelwyatt@microsoft.com>

stas00 committed:
[ds-inference] checkpoint loading => tqdm (#2107)

* [ds-inference] checkpoint loading => tqdm

solve 2 issues:
- less noise using tqdm progress bar
- more informative - tell users how much to wait and how many shards to load

New way:

```
Loading 72 checkpoints:  12%|█▎        | 9/72 [01:12<08:39,  8.25s/it]
```

* write only from one process

* style

jeffra committed:
bump to 0.7.0

Sanger2000 committed:
Dont overwrite hook handles in flop profiler (#2106)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

mrwyattii committed:
use HF NeoX (#2087)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

yaozhewei committed:
Adding DeepSpeed Compression Composer  (#2105)

Co-authored-by: yaozhewei <zheweiy@berkeley.edu>
Co-authored-by: xiaoxiawu <yxiaoxiawu@microsoft.com>
Co-authored-by: Conglong Li <conglong.li@gmail.com>
Co-authored-by: Xiaoxia (Shirley) Wu <94406484+xiaoxiawu-microsoft@users.noreply.github.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>
Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Michael Wyatt <michaelwyatt@microsoft.com>

mrwyattii committed:
Remove hardcoded ROCm install path (#2093)

* fix hard-coded rocm install path

* added fix for newest torch+rocm install

* added backup for not detecting rocm at all

Commit:
fix softmax dim of Residual MoE in moe/layer.py (#2110)

Thanks a lot for finding this issue and fixed it :)
Co-authored-by: Zhewei Yao <zheweiyao@gmail.com>

jeffra committed:
reduce ds-inference log verbosity (#2111)

conglongli committed:
DeepSpeed Compression announcement (#2114)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>
Co-authored-by: Xiaoxia (Shirley) Wu <94406484+xiaoxiawu-microsoft@users.noreply.github.com>

tjruwase committed:
Checkpoint reshaping (#1953)

* unit test, remove exception, add notes

* Move param_shapes to model files

* Remove hard-coded constants

* Conditioned to zero optimizer

* Add zero checkpoint merging

* Print checkpoint version

* Reshape zero_* ckpt files

* Merge zero* files contraction

* Utils for 3D contraction reshaping

* Remove bogus import

* Support bf16_zero ckpts

* Add param slice mappings

* Load universal checkpoints

* Per group mappings from Stas

* Hack to load bf16 zero files

* Param attributes

* WIP

* Fix api bug

* Update lp with local/remote hp

* Disable vocab padding handling

* Update z2 checkpoint

* Remove debug prints

* Remove debug prints; Rebase unit test

* Add reshape assert

* Padding

* Typo

* Catch nonexistent checkpoint path

* Cleanup

* Restore checkpoint state comparisons

* Add torch version guards

* More precise avoidance of false positives.

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

Quentin-Anthony committed:
fix issue in accelerate. (#2121)

Co-authored-by: Ammar Ahmad Awan <ammar.awan@microsoft.com>

jeffra committed:
[docs] website refresh (#2123)

Co-authored-by: Ammar Ahmad Awan <ammar.awan@microsoft.com>
Co-authored-by: yaozhewei <zheweiy@berkeley.edu>
Co-authored-by: Samyam Rajbhandari <samyamr@microsoft.com>

jeffra committed:
[docs] link fixes (#2124)

awan-10 committed:
update info and links. (#2122)

awan-10 committed:
Update moe doc (#2098)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

dependabot[bot] committed:
Bump tzinfo from 1.2.9 to 1.2.10 in /docs (#2126)

Bumps [tzinfo](https://github.com/tzinfo/tzinfo) from 1.2.9 to 1.2.10.
- [Release notes](https://github.com/tzinfo/tzinfo/releases)
- [Changelog](https://github.com/tzinfo/tzinfo/blob/master/CHANGES.md)
- [Commits](https://github.com/tzinfo/tzinfo/compare/v1.2.9...v1.2.10)

---
updated-dependencies:
- dependency-name: tzinfo
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

Quentin-Anthony committed:
DS Benchmarks QoL Improvements (#2120)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jpvillam-amd committed:
[ROCm] Wrong command broke ROCm build. (#2118)

Co-authored-by: Michael Wyatt <michaelwyatt@microsoft.com>

Quentin-Anthony committed:
DeepSpeed Communication Profiling and Logging (#2012)

Co-authored-by: Ammar Ahmad Awan <ammar.awan@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
Delete Gemfile.lock (#2130)

aphedges committed:
Add flake8 to pre-commit checks (#2051)

awan-10 committed:
DS on Azure blog (#2133)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

awan-10 committed:
Azure blog news item (#2135)

awan-10 committed:
Update 2022-07-26-deepspeed-azure.md (#2136)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

yetiansh committed:
Fix conflict between Tutel and top-2 gate in MoE layer (#2053)

* fix conflit between tutel and top-2 gate
Co-authored-by: Ammar Ahmad Awan <ammar.awan@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

pacman100 committed:
adding  HF Accelerate+DS tests workflow (#2134)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

awan-10 committed:
Update 2022-07-26-deepspeed-azure.md (#2138)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
turn off time check for now (#2142)

jeffra committed:
Allow turning off loss scaling wrt GAS + update tput calculator (#2140)

jeffra committed:
[docs] add details about install reqs (#2143)

jeffra committed:
[docs] update build pipeline badges

jeffra committed:
[docs] README formatting fix

jimwu6 committed:
fix broken links (#2144)

mrwyattii committed:
Refactor ZeRO configs to use Pydantic (#2004)

* first pass at pydanticifying Zero Configs

* added pydantic to reqs

* fixed bug with deprecated values not being type-checked

* fixing zero config bugs from unit tests

* fixed access of Config values

* removing zero constants

* formatting/fix broken import

* fixed bad merge

* fixed issue with missing aliased field

* fix for failing tests

* fix how deprecated fields are processed

* only process dep params when they are set

* fix mistyped field name

* fixes, docs, removed more constants

* fix merge

* more fixes after merge w master

* added unit tests

* formatting

* added fix for transformers unit tests

* separated offload config from zero config

* fixed bad import

* formatting and flake fixes

* implement suggestion from review

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

Quentin-Anthony committed:
Add purely-local sliding window sparse attention config (#1962)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
formatting fix for #1962

jeffra committed:
force newer datasets version (#2147)

trajepl committed:
Trajepl/nebula ckpt engine (#2085)

* enable checkpoint engine

* seprated nebula config

* add __init__.py for nebula importing

* linter fix

* fix: ds_config is None

* fix: ds config

* fix: get sd loader fix

* align the API with torch raw code

* linter fix

* remove duplicate tag params

* make checkpoint_engine as required args

* fix args

* extract parameters out to config

* fix: load state dict

* separate load engine

* linter fix

* extract checkpoint engine to abstract calss

* linter fix

* construct function args fix

* add docs for dev/customers

* linter fix

* remove load engine

* print->log_dist

* linter fix

* add tag flag to distinguish the loading order

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Michael Wyatt <michaelwyatt@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

Commit:
Graceful exit on failures for multi-node runs (#2008)

* Use Popen.terminate() to stop the child processes gracefully; Kill them if terminate doesn't work

* The Popen.kill() command cause the training processes to end abruptly. This may cause the child processes to become zombies without communicating properly to the parent process about the kill signal. So the ssh session continue to wait for signals from the child processes, causing it to not return back to the pdsh command

Fixes microsoft#1995

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

shjwudp committed:
fix: fix BF16_Optimizer compatibility issue with optimizer state 0-dim tensor (#2152)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

RezaYazdaniAminabadi committed:
Fix random token-generation issue + MP-checkpoint loading/saving (#2132)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>
Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

ncilfone committed:
Added retain_graph as a kwarg to the main engine backward function (#1149)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

aj-prime committed:
Elastic Training support in DeepSpeed (#2153) (#2156)

Co-authored-by: Arpan Jain <t-arpanjain@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
prevent cuda 10 builds of inference kernels on ampere (#2157)

jeffra committed:
[zero-3] shutdown zero.Init from within ds.init (#2150)

jeffra committed:
enable fp16 input autocasting (#2158)

tjruwase committed:
Release swap buffers for persisted params (#2089)

* Split parameter offload from z3

* Format fixes

* Bug fixes

* Cleanup

* Remove dead code

* Release swap buffers for persisted params

* Format fixes

* Format fixes

* Pass args correctly

* Use pinned memory for nvme offload

* Merge with masster

* Fix missing import

* model pesistence params

* Fix merge issues

* Handle none device

* Usse log_dist

siddharth9820 committed:
Tensor parallelism for Mixture of Experts (#2074)

* tensor parallelism for mixture of experts
Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Ammar Ahmad Awan <ammar.awan@microsoft.com>

jeffra committed:
bump to 0.7.1

mrwyattii committed:
Fix for distributed tests on pytorch>=1.12 (#2141)

Fix for distributed tests
Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

jeffra committed:
delay torch import for inference compatability check (#2167)

zionwu committed:
Fix wrong unit of latency in flops-profiler (#2090) (#2095)

Co-authored-by: Cheng Li <pistasable@gmail.com>
Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

jeffra committed:
[docs] adoption updates (#2173)

mrwyattii committed:
Update for AMD CI workflow (#2172)

re-enable AMD CI with some modifications

jeffra committed:
update offload docs to include stage 1 (#2178)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

RezaYazdaniAminabadi committed:
fixing model partitioning without injection (#2179)

tjruwase committed:
Match compute and reduce dtype (#2145)

* Match compute and reduce dtype

* Unit tests

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

rraminen committed:
Enable fused_lamb_cuda_kernel on ROCm (#2148)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>
Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Michael Wyatt <michaelwyatt@microsoft.com>

hanlint committed:
Update README to latest Composer version (#2177)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

rahilbathwal5 committed:
fix missing import (#2175)

jeffra committed:
[docs] add more models to adoption (#2189)

jeffra committed:
Update nv-lightning-v100.yml (#2190)

gasparitiago committed:
Update README.md (#2192)

Fix typos.

RezaYazdaniAminabadi committed:
Fix the layer-past for GPT based models (#2196)

Dipet committed:
Add gradient_average flag support for sparse grads (#2188)

* Add gradient_average flag support for sparse grads

* formatting fixes

* Add tests

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

minjiaz committed:
Adding additional instructiosn in the compression tutorial on pre-training distillation and quantization for GPT (#2197)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

tjruwase committed:
Log user config exactly (#2201)

RezaYazdaniAminabadi committed:
Fix the tensor-slicing copy for qkv parameters (#2198)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

mrwyattii committed:
Refactor Distributed Tests (#2180)

Refactor Distributed unit tests

kamalkraj committed:
fix table syntax (#2204)

Co-authored-by: Conglong Li <conglong.li@gmail.com>
Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

tjruwase committed:
Correctly detect offload configuration (#2208)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
add cuda 11.7 (#2211)

* add cuda 11.7

* formatting

jeffra committed:
use torch 1.9 (#2215)

awan-10 committed:
[zero-3] print warning once and support torch parameter (#2127)

* print warning only once.

* add support for torch param and only warn on gpu 0

* remove type checking. will be done on a new PR with more tests.

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>
Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

arashb committed:
Add support of OPT models (#2205)

* add opt replace policy

* simplify inf. api

* fix opt replace policy

* fix use-cash & add relu

* Add support of custom MLP act. function

* Revert "simplify inf. api"

This reverts commit 9e910fcbd5471dec9b3c92008426f5ba590bf0b6.

* fix the inference API (temp. solution)

* fix code formatting

* add unit tests for OPT models.

* refactor pre-attention layer norm configuration

* add support of opt-350m model

* refactor the HF model config initialization

* fix hf model config issue

Co-authored-by: Reza Yazdani <reyazda@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>
Co-authored-by: Reza Yazdani <44502768+RezaYazdaniAminabadi@users.noreply.github.com>

zhjohnchan committed:
fix typos in readme. (#2218)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

jeffra committed:
Fix regression w. dist_init_required (#2225)

conglongli committed:
add doc for new bert example (#2224)

RezaYazdaniAminabadi committed:
Remove the random-generator from context during inference (#2228)

* Fix the tensor-slicing copy for qkv parameters

* remove the random-generator from context during inference

* formatting

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
allow saving ckpt w/o ckpt json + bloom copy fix (#2237)

tjruwase committed:
Correctly detect zero_offload (#2213)

* Correctly detect offload configuration

* Correctly detect offload configuration

* Handle deprecated cpu offload setting

* Correcly detect zero_offload setting

* Minor tweak

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>
Co-authored-by: Ammar Ahmad Awan <ammar.awan@microsoft.com>

jeffra committed:
update videos (#2249)

tjruwase committed:
Refactor dist tests: Checkpointing (#2202)

Refactor distributed tests: checkpointing

Co-authored-by: Michael Wyatt <michaelwyatt@microsoft.com>

arashb committed:
Make OPT policy backward compatible with pre-OPT transformers versions (#2254)

RezaYazdaniAminabadi committed:
fix ds-inference without policy (#2247)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
bump to 0.7.2

siddharth9820 committed:
Enable contiguous gradients with Z1+MoE (#2250)

MoE training with zero stage 1 only works with `contiguous gradients=True`.

tjruwase committed:
Correctly detect CPU optimizer usage (#2257)

* Correctly detect CPU optimizer usage

* Update nv-transformers-v100.yml (#2259)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

cmikeh2 committed:
Update half precision header guards (#2261)

yzs981130 committed:
fix #2240: wrong time unit in flops_profiler (#2241)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

jeffra committed:
bump to 0.7.3

mrwyattii committed:
Add blob storage to CI runners (#2260)

Add blob storage to CI runners and enable for transformers cache on inference tests

molly-smith committed:
Update replace_module.py, test-gptj.py related fix (#2269)

Fix RuntimeError: Boolean value of Tensor with more than one value is ambiguous when running test-gptj.py

Dipet committed:
Fix OrderedDict import for python3.6 (#2267)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

RezaYazdaniAminabadi committed:
Ds inference/fix mp2 (#2270)

trajepl committed:
Trajepl: nebula load fix (#2182)

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>
Co-authored-by: chenguo <chenguo@microsoft.com>

jeffra committed:
prevent torch ext folder mkdir at tmp (#2274)

RezaYazdaniAminabadi committed:
Ds-inference Int8 support through ZeroQuant technology (#2217)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

awan-10 committed:
add a new unit test for cuda ops (#2278)

Co-authored-by: cmikeh2 <connorholmes@microsoft.com>

cmikeh2 committed:
Add to codeowners file (#2279)

cmikeh2 committed:
Memory Access Utility (#2276)

Co-authored-by: Ammar Ahmad Awan <ammar.awan@microsoft.com>

RezaYazdaniAminabadi committed:
Fp32 accuracy bug fix (#2285)

Co-authored-by: Arash Bakhtiari <arash@bakhtiari.org>
Co-authored-by: Arash Bakhtiari <arashb@users.noreply.github.com>

tjruwase committed:
Refactor universal checkpointing and tensor fragments (#2253)

* Refactor universal checkpointing and tensor fragments

* Formatting

stas00 committed:
[ds-inference] fix progress bar (#2286)

when loading the non-sharded checkpoint update the progress bar (fix by @RezaYazdaniAminabadi) - I've just tested it to work.

Co-authored-by: Olatunji Ruwase <olruwase@microsoft.com>

tjruwase committed:
Offload all gradients to nvme (#2282)

molly-smith committed:
fused bias relu unittest (#2297)

mrwyattii committed:
fix for pytest picking up local deepspeed dir instead of installed deepspeed (#2299)

Quentin-Anthony committed:
Fix for Zero3 when MP>1 and at least one batch param undefined (#2289)

Co-authored-by: anthony.301 <anthony.301@mri.cluster>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

mrwyattii committed:
Unit test for bias add kernel (#2298)

* added unit test

* Update pt_binding.cpp

* formatting

* Update test_bias_add.py

molly-smith committed:
Update relu.cu with mem_access_utils (#2306)

mrwyattii committed:
Add tensor parallel inference unit tests (#2232)

Co-authored-by: Reza Yazdani <44502768+RezaYazdaniAminabadi@users.noreply.github.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>
Co-authored-by: Sam Ade Jacobs <samjacobs@microsoft.com>

arashb committed:
Fix the residual add mp scaling for  GPTNeoX (#2310)

arashb committed:
Add unit tests for residual_add kernels (#2307)

jeffra committed:
add inference eval scripts (#2303)

mrwyattii committed:
Upgrade P40 tests to torch 1.8 (#2316)

Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

tjruwase committed:
ZeRO-Inference blog (#2271)

* ZeRO-Inference blog

* ZeRO-Inference blog

* Format fixes

* Apply feedback

* Feedback

* Update docs/_posts/2022-08-27-zero-inference.md

Co-authored-by: Stas Bekman <stas00@users.noreply.github.com>

* Update docs/_posts/2022-08-27-zero-inference.md

Co-authored-by: Stas Bekman <stas00@users.noreply.github.com>

* Address feedback

* Format fixes

* More tweaks

* long sequence, nvme offload

* Add image

Co-authored-by: Stas Bekman <stas00@users.noreply.github.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>

tjruwase committed:
ZeRO-Inference blog - wrap up  (#2321)

tjruwase committed:
ZeRO-Inference blog - Update README (#2322)

mrwyattii committed:
refactor to use mem_access (#2317)

GuanhuaWang committed:
add quant unit test (#2315)

* add quant unit test

* add codeowner

* format fix

* fix undefined symbol: curandSetPseudoRandomGeneratorSeed

* modify ref fn name and add comment

* add comments

* add 4bit quant 16groups

* fix

* modify groups in ref code

* parameterize tensor shape

* single param

* detach tensor

* remove -lcurand flag

* add back -lcurand flag

Co-authored-by: Ammar Ahmad Awan <ammar.awan@microsoft.com>

jeffra committed:
only override forward if using cuda-graph (#2291)

mrwyattii committed:
Add more options to inference benchmark (#2325)

jeffra committed:
bump to 0.7.4

samadejacobs committed:
MOE residual matmult unit test (#2323)

MOE residual matmul unit tests

Co-authored-by: Michael Wyatt <michaelwyatt@microsoft.com>
Co-authored-by: Ammar Ahmad Awan <ammar.awan@microsoft.com>

samadejacobs committed:
MOE matmult with memaccess (#2336)

* Fix formatting

* Remove redundant variable

arashb committed:
Refactor residual add kernels (#2333)

Co-authored-by: Ammar Ahmad Awan <ammar.awan@microsoft.com>

GuanhuaWang committed:
mem access for quantize kernel (#2331)

* mem access for quantize kernel

* format

* format fp32

* modify quant kernel

* modify quant kernel2

* modify format

* format

* fix comments in pytest

* fix comments in pytest

* format

* rerun

Co-authored-by: Ammar Ahmad Awan <ammar.awan@microsoft.com>
Co-authored-by: Connor Holmes <connorholmes@microsoft.com>
